app/src/main/java/com/travelcy/travelcy/MainActivity.kt

```diff
@@ -39,6 +52,31 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
+    fun isNetworkConnected(): Boolean {
+        Log.d(TAG, "isNetworkConnected")
+        val capabilities: NetworkCapabilities =
+            connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+            ?: return false
+
+        if (
+            capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET) ||
+            capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) ||
+            capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)
```

Comment:
Duplicate line, should this be something else?

Reply:
You are very right! This should be TRANSPORT_CELLULAR, it would have resulted in isNetworkConnected returning false for all cellular connections, 944edfd fixes this